File: src/main/java/com/example/examplemod/ExampleMod.java
package com.example.examplemod;

import net.fabricmc.api.ModInitializer;
import net.fabricmc.loader.impl.util.log.Log;
import net.fabricmc.loader.impl.util.log.LogCategory;

public class ExampleMod implements ModInitializer {

    public static final String MOD_NAME = "ExampleMod";

    public static final LogCategory LOG_CATEGORY = LogCategory.create(MOD_NAME);

    @Override
    public void onInitialize() {
        Log.info(LOG_CATEGORY, "Initializing " + MOD_NAME);
    }
}

File: src/main/kotlin/com/example/examplemod/ExampleMod.kt
package com.example.examplemod

import net.fabricmc.api.ModInitializer
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class ExampleMod : ModInitializer {
    companion object {
        const val MOD_NAME: String = "ExampleMod"

        val LOGGER: Logger = LoggerFactory.getLogger(MOD_NAME)
    }

    override fun onInitialize() {
        LOGGER.info("Initializing $MOD_NAME with Kotlin")
    }
}
